File: problems_026to050/problem_029/solution.py
import os
import sys
from itertools import product

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None

        integers = range(2, 101)
        terms = set(a**b for (a, b) in product(integers, repeat=2))
        answer = len(terms)

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_029/solution.ts
import { SolutionFunction } from '@/lib/types';
import { range } from '@/lib/toolbox';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = null;

  const integers = range(2, 101);
  const terms: Set<number> = new Set();

  integers.forEach((a) => {
    integers.forEach((b) => {
      terms.add(a ** b);
    });
  });

  answer = terms.size;

  return answer;
};

export default solution;
